### PR TITLE
Add PostgreSQL as a possible viewer

### PR DESCRIPTION
**Pull request by cevian**
Add documentation for how to use PostgreSQL with pgai to access the dataset.

### CODE REVIEW

File: docs/source/_toctree.yml [hunk 1]
```diff
@@ -44,6 +44,8 @@
           title: Pandas
         - local: polars
           title: Polars
+        - local: postgresql
+          title: PostgreSQL
```
Comment (julien-c) [inline docs/source/_toctree.yml:48]: can you also add the link inside `parquet_process.md`?

File: docs/source/postgresql.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Alternatively, you can install pgai into an existing PostgreSQL database. For instructions on how to install pgai into an existing PostgreSQL database, follow the instructions in the [github repo](https://github.com/timescale/pgai).
+
+## Create a table from a dataset
+
+To load a dataset into PostgreSQL, you can use the `ai.load_dataset` function. This function will create a PostgreSQL table, and load the dataset from the Hugging Face Hub
+in a streaming fashion.
+
+```sql
+select ai.load_dataset('squad');
```
Comment (julien-c) [inline docs/source/postgresql.md:24]:
```suggestion
select ai.load_dataset('rajpurkar/squad');
```

does it also work with namespaced names? (all datasets on HF are namespaced now)

**Review by julien-c — COMMENTED** [review]
cool addition! 

could make sense to add it to https://huggingface.co/docs/hub/datasets-libraries too in the future, maybe

**Comment by cevian** [comment]
@julien-c addressed your comments. Yeah https://huggingface.co/docs/hub/datasets-libraries would be cool. Do you think we should try to get Push to hub abilities in before we do that?

**Comment by julien-c** [comment]
Push to hub would be awesome! do you see some demand for this from `pgai` users?

**Comment by cevian** [comment]
Not yet. But we've just added load_datasets so I think it will take people a bit to catch on.

**Review by julien-c — APPROVED** [review]
ok pinging @lhoestq for a review but this PR lgtm

**Review by lhoestq — APPROVED** [review]
Great ! Also interested in a push to hub feature, but not need for it to start sharing about this online imo !

Let me know if you post about it online I can re-share with the community :)

Btw I've seen people interested in RAG applications for code documentation lately, let me know if you ever do a demo on this topic

> could make sense to add it to https://huggingface.co/docs/hub/datasets-libraries too in the future, maybe

We might re-organize this indeed, it's a bit suprising to me to have libraries in the Viewer's docs compared to libraries integrated with the Hub directly (especially for tools like `pgai` that provide deep integrations). Anyway I'll keep you posted if we move this page to make it more visible :)

**Comment by cevian** [comment]
@lhoestq we shared this here: [X post](https://x.com/avthars/status/1869842858834079916) and on [linkedIn](https://www.linkedin.com/posts/avthars_postgresql-huggingface-opensource-activity-7275609557299523584-3_Vq?utm_source=share&utm_medium=member_desktop) and on [bsky](https://bsky.app/profile/ceviantech.bsky.social/post/3ldorraoppc27). Thanks for re-sharing,

I'll let you know about demos involving code documentation for sure.

**Comment by lhoestq** [comment]
Awesome ! glad to see it's getting good reception as well :) 

> I'll let you know about demos involving code documentation for sure.

ok !

**Pull request merged**